src/hardware/driver/ltc2320.rs

```diff
@@ -0,0 +1,137 @@
+///! LTC2320 Driver
+///!
+///! QSPI bug (2.4.3):
+///! https://www.st.com/resource/en/errata_sheet/es0392-stm32h742xig-and-stm32h743xig-device-limitations-stmicroelectronics.pdf
+///!
+///! This driver is intended to be used in the following manner:
+///! 1. Trigger a new LTC2320 conversion with start_conversion(). This sets nCNV low and waits until TCONV has passed.
+///!    Then it pases the timer and starts the qspi readout.
+///! 2. Call handle_transfer_done_irq() in the QSPI ISR to retrieve the ADC data and set nCNV high again.
+///!
+///! Only works under the following condition:
+///! Conversions are not restarted faster than (T_readout + TCONV + TCNVH + readout/irq CPU overhead).
+///! You will always see an error
+use super::super::hal::{
+    device::QUADSPI,
+    gpio::{self, gpiob, gpioc, gpioe},
+    prelude::*,
+    rcc, stm32,
+    time::NanoSeconds,
+    timer::Timer,
+    xspi::{Qspi, QspiError, QspiMode, XspiExt},
+};
+use core::ptr;
+use fugit::Hertz;
+use nb::block;
+
+#[derive(Copy, Clone, Debug)]
+pub enum Error<E> {
+    TimerBusyError,
+    QspiError(E),
+}
+
+impl<E> From<E> for Error<E> {
+    fn from(err: E) -> Error<E> {
+        Error::QspiError(err)
+    }
+}
+
+pub struct Ltc2320Pins {
+    pub spi: (
+        gpiob::PB2<gpio::Alternate<9>>,
+        gpioe::PE7<gpio::Alternate<10>>,
+        gpioe::PE8<gpio::Alternate<10>>,
+        gpioe::PE9<gpio::Alternate<10>>,
+        gpioe::PE10<gpio::Alternate<10>>,
+    ),
+    pub cnv: gpioc::PC11<gpio::Output<gpio::PushPull>>,
+}
+
+pub struct Ltc2320 {
+    qspi: Qspi<QUADSPI>,
+    cnv: gpioc::PC11<gpio::Output<gpio::PushPull>>,
+    timer: Timer<stm32::TIM7>,
+}
+
+impl Ltc2320 {
+    const N_BYTES: usize = 16; // Number of bytes to be transfered.
+    const TCONV: NanoSeconds = NanoSeconds::from_ticks(450u32); // minimum conversion time according to datasheet
+    pub fn new(
+        clocks: &rcc::CoreClocks,
+        qspi_rec: rcc::rec::Qspi,
+        qspi_peripheral: stm32::QUADSPI,
+        qspi_frequency: Hertz<u32>,
+        timer_rec: rcc::rec::Tim7,
+        timer_peripheral: stm32::TIM7,
+        timer_frequency: Hertz<u32>,
+        mut pins: Ltc2320Pins,
+    ) -> Self {
+        let mut qspi =
+            qspi_peripheral.bank2(pins.spi, qspi_frequency, clocks, qspi_rec);
+        qspi.configure_mode(QspiMode::OneBit).unwrap();
+        qspi.is_busy().unwrap(); // panic if qspi busy
```

Comment:
Are you sure? I'd expect a Result<bool, _>

Reply:
nope, returns an Err(Busy) if busy

Reply:
It's a `Result<(), Busy>` which is a bit counter the usual API naming schemes.
Hence this is fine.